README.md

```diff
@@ -0,0 +1,211 @@
+# Folder app
+
+The goal is to create a safe web application for browsing folders.
+
+- [Folder app](#folder-app)
+  - [Frontend](#frontend)
+    - [Assets server](#assets-server)
+    - [Tooling](#tooling)
+    - [Routes](#routes)
+      - [/](#)
+      - [/login](#login)
+      - [/not-found](#not-found)
+      - [/folder/\*\*/\*](#folder)
+    - [Design](#design)
+  - [Backend](#backend)
+    - [Tooling](#tooling-1)
+    - [API](#api)
+      - [GET /folder/\*\*/\*](#get-folder)
+      - [POST /session](#post-session)
+      - [DELETE /session](#delete-session)
+  - [Security considerations](#security-considerations)
+  - [Branch naming convention](#branch-naming-convention)
+
+## Frontend
+
+Typical single page application.
+
+### Assets server
+
+> SAFETY NOTE:
+> The server should serve assets via [HTTPS](https://en.wikipedia.org/wiki/HTTPS).
+> I will not implement that.
+
+![spa](/images/spa.png)
+
+### Tooling
+
+- [react](reactjs.org)
+- [react-router](https://reactrouter.com/)
+- [typescript](https://www.typescriptlang.org/)
+- [styled-components](https://styled-components.com/)
+- [vite](https://vitejs.dev/)
+- [testing-library](https://testing-library.com/)
+- [nginx](https://www.nginx.com/)
+- [docker](https://www.docker.com/)
+
+### Routes
+
+#### /
+
+Redirects user to `/folder`.
+
+#### /login
+
+A login screen where an unauthenticated user is automatically redirected to (and then taken back to the original URL).
+
+#### /not-found
+
+Not found page, where a user is automatically redirected to in case folder, does not exist.
+
+#### /folder/\*\*/\*
+
+The page displays single folder content.
+
+### Design
+
+See [figma](https://www.figma.com/file/J6yvOILo6HM62FnHXaAAOC/FolderApp?node-id=0%3A1)
+
+- Based on google drive (material design).
+- Icons by [Icons8](https://icons8.com/)
+
+## Backend
+
+Folder data service. In-memory session authentication.
+
+### Tooling
+
+- [rust](https://www.rust-lang.org/)
+- [rocket](https://rocket.rs/)
+- [serde](https://serde.rs/)
+- [docker](https://www.docker.com/)
+
+### API
+
+> SAFETY NOTE:
+> In case we will use real DB to store data we should validate input against some kind of query injection (like SQL injection).
+
+#### GET /folder/\*\*/\*
```

Comment:
No, this will be impossible: `/**/*` is always an absolute path (relative operators will not be supported). 
There is the case where `GET /folder/../..` will result in folder structure: root will contain a directory named `..` and inside of it there will be a directory named `..`.

Reply:
This is rather interesting.
The browser will translate `myApp.io/folder/foo/../bar` into `myApp.io/folder/bar`.
https://www.rfc-editor.org/rfc/rfc3986#section-5.2.4